src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -354,29 +358,18 @@ internal bool CheckValueKind(SyntaxNode node, BoundExpression expr, BindValueKin
                 case BoundKind.ArrayAccess:
                 case BoundKind.PointerIndirectionOperator:
                 case BoundKind.PointerElementAccess:
-                    // array elements and pointer dereferencing are readwrite varaibles
+                    // array elements and pointer dereferencing are readwrite variables
                     return true;
 
                 case BoundKind.RefValueOperator:
                     // The undocumented __refvalue(tr, T) expression results in a variable of type T.
-                    // it is a readwrite variable, but could refer to local data
-                    if (!RequiresReturnableReference(valueKind))
```

Comment:
It looks like no test change corresponds to this code change. Can you point me to some relevant tests or add some? Or was the old code superfluous?

Reply:
We continue not allowing ref-returning __refvalue. For that we have tests. It may make sense to add tests that __refvalue cannot ref-escape the scope, but without ref reassingment it would not be testable. The only thing that can be added here is that a Span that wraps a __refvalue cannot value-escape from its scope.
I will add a test for that.

---
In reply to: [135400976](https://github.com/dotnet/roslyn/pull/21334#discussion_r135400976) [](ancestors = 135400976)

Reply:
I see that the PR was merged. Will you add this test in a subsequent PR?

---
In reply to: [135665037](https://github.com/dotnet/roslyn/pull/21334#discussion_r135665037) [](ancestors = 135665037,135400976)

Reply:
there are couple issues around handling of __refvalue that are NYI and marked with //PROTOTYPE. The implementation may change once I get to those, but there will definitely be tests. 
I had to fix couple pretty bad bugs in this code, so did not have time yet.